### PR TITLE
Check logic - Ready For Review

### PR DESCRIPTION
**Pull request by AlekstheRealist**
Code is finished. Waiting for valid_move methods to get merged so the code can work. If there's any other ideas for tests let us know

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -61,4 +63,30 @@ def make_queens!
     Queen.create(game_id: self.id, color: 'white', x_cord: 3, y_cord: 0)
     Queen.create(game_id: self.id, color: 'black', x_cord: 3, y_cord: 7)
   end
+
+  def check?
+    black_king_in_check? || white_king_in_check?
+  end
+
+  def black_king_in_check?
```
Comment (ilyakrasnov) [inline app/models/game.rb]: Just as an exercise: 

Could we merge the two methods (black_king_in_check? and white_king_in_check?) in one? They seem pretty identical..

File: db/migrate/20160709205224_add_check_to_games.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddCheckToGames < ActiveRecord::Migration
+  def change
+    add_column :games, :check, :string
```
Comment (ilyakrasnov) [inline db/migrate/20160709205224_add_check_to_games.rb]: Do you think we need it? Whats the reason to save it in the database? Its a very dynamic state, isn't it?

File: db/migrate/20160712003303_remove_check_from_game.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class RemoveCheckFromGame < ActiveRecord::Migration
+  def change
+    remove_column :games, :check
```
Comment (ilyakrasnov) [inline db/migrate/20160712003303_remove_check_from_game.rb]: Oops,  taking my previous comment back 😄

File: app/models/game.rb [hunk 4]
```diff
@@ -16,49 +16,92 @@ def initialize_board!
...
+    Queen.create(game_id: id, color: 'white', x_cord: 3, y_cord: 0)
+    Queen.create(game_id: id, color: 'black', x_cord: 3, y_cord: 7)
   end
+
+  # def check?
+  #   black_king_in_check? || white_king_in_check?
+  # end
+
+  def check?
+    king = King.where(game_id: id) # ==> king = [black king, white king]
```
Comment (ilyakrasnov) [inline app/models/game.rb]: lets name them *kings* 😉

File: app/models/game.rb [hunk 5]
```diff
@@ -16,49 +16,92 @@ def initialize_board!
...
+    actual_piece = Piece.where(game_id: id)
```
Comment (ilyakrasnov) [inline app/models/game.rb]: again lets name it *actual_pieces*

**Comment by ruishine** [comment]
aha!

File: app/models/game.rb [hunk 6]
```diff
@@ -19,50 +19,63 @@ def initialize_board!
...
+  end
+
+  def check?
+    kings = King.where(game_id: id)
+    kings.each do |king|
+      pieces.each do |piece|
+        return true if piece.color != king.color && piece.valid_move?(king.x_cord, king.y_cord)
+      end
+    end
+
```
Comment (ruishine) [inline app/models/game.rb]: really this was it!?!!!

**Comment by ruishine** [comment]
if we are not waiting for valid_moves to be changed anymore and rspecs pass... thumbs up!!

**Comment by ilyakrasnov** [comment]
👍

**Pull request merged**